server/lib/scaletozero/scaletozero.go

```diff
@@ -0,0 +1,64 @@
+package scaletozero
+
+import (
+	"context"
+	"os"
+
+	"github.com/onkernel/kernel-images/server/lib/logger"
+)
+
+// Unikraft scale-to-zero control file
+// https://unikraft.cloud/docs/api/v1/instances/#scaletozero_app
+const unikraftScaleToZeroFile = "/uk/libukp/scale_to_zero_disable"
+
+type Controller interface {
+	// Disable turns scale-to-zero off.
+	Disable(ctx context.Context) error
+	// Enable re-enables scale-to-zero after it has previously been disabled.
+	Enable(ctx context.Context) error
+}
+
+type unikraftCloudController struct {
+	path string
+}
+
+func NewUnikraftCloudController() Controller {
+	return &unikraftCloudController{path: unikraftScaleToZeroFile}
+}
+
+func (c *unikraftCloudController) Disable(ctx context.Context) error {
+	return c.write(ctx, "+")
+}
+
+func (c *unikraftCloudController) Enable(ctx context.Context) error {
+	return c.write(ctx, "-")
+}
+
+func (c *unikraftCloudController) write(ctx context.Context, char string) error {
+	if _, err := os.Stat(c.path); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		logger.FromContext(ctx).Error("failed to stat scale-to-zero control file", "path", c.path, "err", err)
+		return err
+	}
+
+	f, err := os.OpenFile(c.path, os.O_WRONLY|os.O_TRUNC, 0)
+	if err != nil {
+		logger.FromContext(ctx).Error("failed to open scale-to-zero control file", "path", c.path, "err", err)
+		return err
+	}
+	defer f.Close()
+	if _, err := f.Write([]byte(char)); err != nil {
+		logger.FromContext(ctx).Error("failed to write scale-to-zero control file", "path", c.path, "err", err)
+		return err
+	}
+	return nil
+}
```

Comment:
see https://unikraft.cloud/docs/api/v1/instances/#scale-to-zero:~:text=Unikraft%20Cloud%20allows,file%2Dbased%20interface%3A